src/main/java/uk/gov/hmcts/reform/slc/logging/AppInsights.java

```diff
@@ -0,0 +1,134 @@
+            success
+        );
+    }
+
+    public void trackMessageDeadLetteredInServiceBus(java.time.Duration duration, boolean success) {
+        telemetry.trackDependency(
+            AppDependency.SERVICE_BUS,
+            AppDependencyCommand.SERVICE_BUS_DEAD_LETTERED,
+            new Duration(duration.toMillis()),
+            success
+        );
+    }
+
+    public void trackPdfGenerator(java.time.Duration duration, boolean success) {
+        telemetry.trackDependency(
+            AppDependency.PDF_GENERATOR,
+            AppDependencyCommand.GENERATE_PDF_FROM_HTML,
+            new Duration(duration.toMillis()),
+            success
+        );
+    }
+
+    public void trackFtpUpload(java.time.Duration duration, boolean success) {
+        telemetry.trackDependency(
+            AppDependency.FTP_CLIENT,
+            AppDependencyCommand.FTP_FILE_UPLOAD,
+            new Duration(duration.toMillis()),
+            success
+        );
+    }
+
+    // EVENTS
+
+    public void trackMessageReceived(String messageId, long enqueuedInNanos) {
+        telemetry.trackEvent(
+            AppEvent.MESSAGE_RECEIVED,
+            Collections.singletonMap("messageId", messageId),
+            Collections.singletonMap("enqueuedInNanos", (double) enqueuedInNanos));
+    }
+
+    public void markMessageHandled(String messageId, long handlingInNanos) {
+        telemetry.trackEvent(
+            AppEvent.MESSAGE_HANDLED_SUCCESSFULLY,
+            Collections.singletonMap("messageId", messageId),
+            Collections.singletonMap("handledInNanos", (double) handlingInNanos)
+        );
+    }
+
+    public void markMessageNotHandled(String messageId, long handlingInNanos) {
+        telemetry.trackEvent(
+            AppEvent.MESSAGE_HANDLED_UNSUCCESSFULLY,
+            Collections.singletonMap("messageId", messageId),
+            Collections.singletonMap("handledInNanos", (double) handlingInNanos)
+        );
+    }
+
+    public void trackMessageMappedToLetter(String messageId, String serviceName, String template, long bodyLength) {
+        Map<String, String> properties = new HashMap<>();
+
+        properties.put("messageId", messageId);
+        properties.put("service", serviceName);
+        properties.put("template", template);
+
+        telemetry.trackEvent(
+            AppEvent.MESSAGE_MAPPED_SUCCESSFULLY,
+            properties,
+            Collections.singletonMap("messageSize", (double) bodyLength)
+        );
+    }
+
+    public void trackMessageMappedToNull(String messageId, long bodyLength) {
+        telemetry.trackEvent(
+            AppEvent.MESSAGE_MAPPED_EMPTY,
+            Collections.singletonMap("messageId", messageId),
+            Collections.singletonMap("messageSize", (double) bodyLength)
+        );
+    }
+
+    public void trackMessageMappedToInvalid(String messageId, long bodyLength) {
+        telemetry.trackEvent(
+            AppEvent.MESSAGE_MAPPED_INVALID,
```

Comment:
is this an event??

Reply:
event of invalid letter body

Reply:
I'd think it's an error which should be tracked with `tracker.trackException`

Reply:
i left exception tracking for third party errors. letter mapping is built in feature and cover each case as an event. we throw our exception when reaching particular part